### ISSUE TITLE
Drop dependency to vin_decoder

### ISSUE THREAD

**Issue by knaeckeKami** [dart-league/validators#26]
I would like to propose that this package should remove its dependency on vin_decoder.
My reasoning is:

- it looks like it is only used for this simple function:
```dart
/// check if the string is a valid ISO 3779-compliant Vehicle identification
/// number (VIN)
bool isVIN(String str) {
  VIN vin = VIN(number: str);
  return vin.valid();
}
``` 
which could be easily pasted by the users who need it
- `validators` is used by many other packages
- `vin_decoder` has many dependencies, e.g. to `http` , `basic_utils`  and `random_string`  which are also used by many other packages
- therefore, it is likely to run into dependency conflicts because of all these transitive dependencies
- I think the VIN feature is only used by a tiny fraction of users, but many are currently having issues with dependencies due to the migration to null safety

This would be a breaking change, but IMO it would be worthwhile. validators would then be a package without any dependencies.

**Comment by ChristopherBrand** [comment]
I agree. I really need to move on to null safety. All the firebase packages are moving forward and I can't afford to get left too far behind.

**Comment by jibiel** [comment]
I second that.

Every other package update stumbles over this `vin_decoder` problem, which can't be resolved with simple `dependency_overrides` declaration (which is nasty workaround anyway), because of its dependencies (which, in turn, have even more dependencies, and so on).

Another solution would be dropping dependency to `validators` altogether and using [`string_validator`](https://pub.dev/packages/string_validator), which has less cooler name and has issue tracker disabled on [GitHub](https://github.com/suragch/string_validator), but has no dependencies, supports null safety and forks the same repo (meaning it has almost exactly the same API).

**Comment by zeusbaba** [comment]
I totally agree with this proposal.   
Therefore, I just created a [pull-request](https://github.com/dart-league/validators/pull/27)  
@luisvt if you need help in maintaining & publishing this package, plz let me know.  

in the meantime, if anyone wants to use my fork, https://github.com/beerstorm-net/validators  
```
validators:
    git: git://github.com/beerstorm-net/validators.git
```

**Comment by knaeckeKami** [comment]
this was merged in #27
